### ISSUE TITLE
Builds fail for node v10

### ISSUE THREAD

**Issue by alrra** [sonarwhal/sonarwhal#1027]
#### 1) `upath`  (fixed by https://github.com/sonarwhal/sonarwhal/commit/4cafaa626b716834a4f8ade182425e13510fbbd5)

Currently because of `upath` (which down the line is used by `cpx`) [`yarn install` won't work]( https://travis-ci.org/sonarwhal/sonarwhal/jobs/370914636#L474) when `node` `v10` is used.

Possible workaround: use [`yarn --ignore-engines`](https://yarnpkg.com/lang/en/docs/cli/install/#toc-yarn-install-ignore-engines)

---

See also:

  * https://github.com/anodynos/upath/issues/14
  * https://github.com/anodynos/upath/pull/15

---

#### 2) `canvas-prebuilt`

`canvas-prebuilt` also fails

```
canvas-prebuilt: node-pre-gyp http GET https://github.com/chearon/node-canvas-prebuilt/releases/download/v1.6.5-prerelease.1
error /Users/xxxxx/projects/sonarwhal/node_modules/canvas-prebuilt: Command failed.
Exit code: 1
Command: node_modules/.bin/node-pre-gyp install
Arguments: 
Directory: /Users/xxxxx/projects/sonarwhal/node_modules/canvas-prebuilt
Output:
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@0.6.39
node-pre-gyp info using node@10.0.0 | darwin | x64
node-pre-gyp info check checked for "/Users/xxxxx/projects/sonarwhal/node_modules/canvas-prebuilt/canvas/build/Release/canvas-prebuilt.node" (not found)
node-pre-gyp http GET https://github.com/chearon/node-canvas-prebuilt/releases/download/v1.6.5-prerelease.1/canvas-prebuilt-v1.6.5-prerelease.1-node-v64-darwin-x64.tar.gz
node-pre-gyp http 404 https://github.com/chearon/node-canvas-prebuilt/releases/download/v1.6.5-prerelease.1/canvas-prebuilt-v1.6.5-prerelease.1-node-v64-darwin-x64.tar.gz
node-pre-gyp ERR! install error 
node-pre-gyp ERR! stack Error: 404 status code downloading tarball https://github.com/chearon/node-canvas-prebuilt/releases/download/v1.6.5-prerelease.1/canvas-prebuilt-v1.6.5-prerelease.1-node-v64-darwin-x64.tar.gz
node-pre-gyp ERR! stack     at Request.<anonymous> (/Users/xxxxx/projects/sonarwhal/node_modules/node-pre-gyp/lib/install.js:118:27)
node-pre-gyp ERR! stack     at Request.emit (events.js:187:15)
node-pre-gyp ERR! stack     at Request.onRequestResponse (/Users/xxxxx/projects/sonarwhal/node_modules/node-pre-gyp/node_modules/request/request.js:1074:10)
node-pre-gyp ERR! stack     at ClientRequest.emit (events.js:182:13)
node-pre-gyp ERR! stack     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:534:21)
node-pre-gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:109:17)
node-pre-gyp ERR! stack     at TLSSocket.socketOnData (_http_client.js:421:20)
node-pre-gyp ERR! stack     at TLSSocket.emit (events.js:182:13)
node-pre-gyp ERR! stack     at addChunk (_stream_readable.js:280:12)
node-pre-gyp ERR! stack     at readableAddChunk (_stream_readable.js:265:11)
node-pre-gyp ERR! System Darwin 17.5.0
node-pre-gyp ERR! command "/Users/xxxxx/.nvm/versions/node/v10.0.0/bin/node" "/Users/alrra/projects/sonarwhal/node_modules/canvas-prebuilt/node_modules/.bin/node-pre-gyp" "install"
node-pre-gyp ERR! cwd /Users/xxxxx/projects/sonarwhal/node_modules/canvas-prebuilt
node-pre-gyp ERR! node -v v10.0.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.39
```

---

- [x] `upath` 
- [ ] `canvas-prebuilt`

**Comment by alrra** [comment]
2) `canvas-prebuilt`

`canvas-prebuilt` also fails

```
canvas-prebuilt: node-pre-gyp http GET https://github.com/chearon/node-canvas-prebuilt/releases/download/v1.6.5-prerelease.1
error /Users/xxxxx/projects/sonarwhal/node_modules/canvas-prebuilt: Command failed.
Exit code: 1
Command: node_modules/.bin/node-pre-gyp install
Arguments: 
Directory: /Users/xxxxx/projects/sonarwhal/node_modules/canvas-prebuilt
Output:
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@0.6.39
node-pre-gyp info using node@10.0.0 | darwin | x64
node-pre-gyp info check checked for "/Users/xxxxx/projects/sonarwhal/node_modules/canvas-prebuilt/canvas/build/Release/canvas-prebuilt.node" (not found)
node-pre-gyp http GET https://github.com/chearon/node-canvas-prebuilt/releases/download/v1.6.5-prerelease.1/canvas-prebuilt-v1.6.5-prerelease.1-node-v64-darwin-x64.tar.gz
node-pre-gyp http 404 https://github.com/chearon/node-canvas-prebuilt/releases/download/v1.6.5-prerelease.1/canvas-prebuilt-v1.6.5-prerelease.1-node-v64-darwin-x64.tar.gz
node-pre-gyp ERR! install error 
node-pre-gyp ERR! stack Error: 404 status code downloading tarball https://github.com/chearon/node-canvas-prebuilt/releases/download/v1.6.5-prerelease.1/canvas-prebuilt-v1.6.5-prerelease.1-node-v64-darwin-x64.tar.gz
node-pre-gyp ERR! stack     at Request.<anonymous> (/Users/xxxxx/projects/sonarwhal/node_modules/node-pre-gyp/lib/install.js:118:27)
node-pre-gyp ERR! stack     at Request.emit (events.js:187:15)
node-pre-gyp ERR! stack     at Request.onRequestResponse (/Users/xxxxx/projects/sonarwhal/node_modules/node-pre-gyp/node_modules/request/request.js:1074:10)
node-pre-gyp ERR! stack     at ClientRequest.emit (events.js:182:13)
node-pre-gyp ERR! stack     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:534:21)
node-pre-gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:109:17)
node-pre-gyp ERR! stack     at TLSSocket.socketOnData (_http_client.js:421:20)
node-pre-gyp ERR! stack     at TLSSocket.emit (events.js:182:13)
node-pre-gyp ERR! stack     at addChunk (_stream_readable.js:280:12)
node-pre-gyp ERR! stack     at readableAddChunk (_stream_readable.js:265:11)
node-pre-gyp ERR! System Darwin 17.5.0
node-pre-gyp ERR! command "/Users/xxxxx/.nvm/versions/node/v10.0.0/bin/node" "/Users/alrra/projects/sonarwhal/node_modules/canvas-prebuilt/node_modules/.bin/node-pre-gyp" "install"
node-pre-gyp ERR! cwd /Users/xxxxx/projects/sonarwhal/node_modules/canvas-prebuilt
node-pre-gyp ERR! node -v v10.0.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.39
```

**Comment by alrra** [comment]
Note: I made [node v10  builds be allowed to fail for now](https://github.com/sonarwhal/sonarwhal/commit/36b80b21f9f955448f7fbe692a6f797f0e6d4b9b) in order to still allow the site and other things to be updated.

**Comment by molant** [comment]
I've opened an issue for `canvas-prebuilt`: https://github.com/node-gfx/node-canvas-prebuilt/issues/36

**Comment by alrra** [comment]
> upath

[⬆️ will soon be fixed](https://github.com/anodynos/upath/pull/15#issuecomment-385045789)!

**Comment by alrra** [comment]
> https://github.com/anodynos/upath/pull/15

Has been merged! 🎉

**Comment by anodynos** [comment]
Hello everyone - I merged [anodynos/upath#15](https://github.com/anodynos/upath/pull/15) & published it  - thanks & sorry for the delay! You can close the issue, once you check its working. Sorry for the delay ;-)

**Comment by alrra** [comment]
> I merged anodynos/upath#15 & published it 

Thank you, @anodynos! We sincerely appreciate it! 💜

**Comment by alrra** [comment]
`upath` part is now fixed by https://github.com/sonarwhal/sonarwhal/commit/4cafaa626b716834a4f8ade182425e13510fbbd5.
